lib/Github/HttpClient/Cache/ResponseSerializer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Github\HttpClient\Cache;
+
+use Http\Discovery\StreamFactoryDiscovery;
+use Http\Message\StreamFactory;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Serialize and unserialize a PSR-7 response.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ResponseSerializer
+{
+    /**
+     * @param ResponseInterface $response
+     * @param StreamFactory|null $streamFactory
+     *
+     * @return array
+     */
+    public static function serialize(ResponseInterface $response, StreamFactory $streamFactory = null)
+    {
+        $streamFactory = $streamFactory ?: StreamFactoryDiscovery::find();
+
+        $bodyStream = $response->getBody();
+        $body = $bodyStream->__toString();
+        if ($bodyStream->isSeekable()) {
+            $bodyStream->rewind();
+        } else {
+            $response = $response->withBody($streamFactory->createStream($body));
```

Comment:
This is a weird hack.. If the stream is not seekable we can not rewind it. This could mean that we are only allowed toread it once and can't read it again. That's why we have replace the Response with a new Stream so others can read it after us.